packages/store/dev/components/CounterButton.tsx

```diff
@@ -0,0 +1,60 @@
+import type { ParentComponent, ComponentProps, JSX, JSXElement } from "solid-js";
+import { splitProps } from "solid-js";
+import type { CounterState, CounterActions } from "../stores/counter-store";
+import { counterStore } from "../stores/counter-store";
+
+export type CounterButtonProps = Omit<ComponentProps<"button">, "onClick">;
+
+export type CounterButtonType = ParentComponent<
+  CounterButtonProps & {
+    onClick: (state: CounterState, actions: CounterActions) => void;
+    text?: string;
+  }
+>;
+
+export const CounterButton: CounterButtonType = props => {
+  const { state: counterState, actions: counterActions } = counterStore;
+  const [counterProps, buttonProps] = splitProps(props, ["onClick", "text"]);
+  const onClick = () => counterProps.onClick(counterState, counterActions);
+
+  return (
+    <>
+      <button {...buttonProps} onClick={onClick}>
+        {!!props.text ? props.text : props.children}
+      </button>
+    </>
+  );
+};
+export default CounterButton;
+
+export const IncreaseButton: ParentComponent<CounterButtonProps> = props =>
+  CounterButton({
+    ...props,
+    onClick: (state, actions) => {
+      actions.resetCount(state.value + 1);
+    },
+    text: "+"
+  });
```

Comment:
I know this is only a demo, but I wouldn't add code like this, just because someone might copy it and cause some unwanted side-effects.
In solid there is little difference between `<Component/>` and `Component()`, but one very important detail is that the function call should be untracked to avoid tracking signals read in the component body.
```suggestion
  <CounterButton
    {...props}
    onClick={(state, actions) => {
      actions.resetCount(state.value + 1);
    }}
    text="+"
  />
```

Reply:
I had wondered what the difference was between what I had and using `createComponent` but I didn't look it up until just now. Thanks for fixing this.